### PR TITLE
Some needed stuff

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
run tests

File: requirements/runtime.txt [hunk 1]
```diff
@@ -5,12 +5,12 @@ numpy
...
-torch>=1.3
-torchvision
+torch==1.5.1
+torchvision==0.6.1
 pytorchcv==0.0.55
 tqdm
-onnx>=1.6.0
-onnxruntime>=1.1.2
+onnx==1.6.0
+onnxruntime==1.1.2
```
Comment (druzhkov-paul) [inline requirements/runtime.txt]: Why is this?

File: tools/train.py [hunk 2]
```diff
@@ -215,11 +218,23 @@ def main():
...
+    if distributed:
+        dataset_len_per_gpu = dataset_len_per_gpu // get_dist_info()[1]
+    assert dataset_len_per_gpu > 0
     if cfg.data.samples_per_gpu == 'auto':
-        cfg.data.samples_per_gpu = determine_max_batch_size(cfg, distributed)
+        logger.info(f'Auto-selection of samples per gpu (batch size).')
+        cfg.data.samples_per_gpu = determine_max_batch_size(cfg, distributed, dataset_len_per_gpu)
+        logger.info(f'Auto selected batch size: {cfg.data.samples_per_gpu} {dataset_len_per_gpu}')
         cfg.dump(osp.join(cfg.work_dir, osp.basename(args.config)))
+    if dataset_len_per_gpu < cfg.data.samples_per_gpu:
```
Comment (druzhkov-paul) [inline tools/train.py:232]: Just a note. If the dataset is that small, it may be a better choice to repeat it several times, rather than decrease the batch size. Though it'll require training schedule correction as well, making this a bit tricky. On the other hand training speed is  among the pros of extending the dataset, 'cause there is an overhead on starting a new epoch.

File: requirements/runtime.txt (hunk 1 above)
Comment (Ilya-Krylov) [reply requirements/runtime.txt]: There were several times when code becomes non-working because of new version of pytorch (and probably onnx). I've decided to fix onnx stuff as well. Do you think it is a bad idea just to fix requirements?

File: tools/train.py (hunk 2 above)
Comment (Ilya-Krylov) [reply tools/train.py:232]: To my mind it should not be the case in normal situation. Probably it is good enough workaround to see that training is going well with pre-defined configuration files from OTE and user-collected very small dataset. I would leave it as is, though your comments are reasonable for sure.

File: requirements/runtime.txt (hunk 1 above)
Comment (druzhkov-paul) [reply requirements/runtime.txt]: IMO it's more of a workaround than a solution, and it would be good to have more confidence in what have caused those issues. This being said, if latest versions of onnx and onnxruntime really make some troubles here, I don't mind fixing it this was.

Comment (Ilya-Krylov) [reply requirements/runtime.txt]: Why >= is actually needed? For most packages we use ==. With >= we always have a chance that something goes wrong and we may not even know about it.

Comment (druzhkov-paul) [reply requirements/runtime.txt]: And we won't have any advantage on the latest features, fixes and improvements at the same time. It can also make the whole project less user-friendly. I believe that possible regressions should be handled at a CI side. But as long as we don't have the ready to use infrastructure for that, then I don't mind disabling versions' flexibility.

**Comment by Ilya-Krylov** [comment]
run tests

File: requirements/runtime.txt (hunk 1 above)
Comment (Ilya-Krylov) [reply requirements/runtime.txt]: reverted flexibility, currently it does not create problems. Please "Approve" if you are ok with changes, as soon as CI is green.

**Review by druzhkov-paul — APPROVED** [review]
